zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala

```diff
@@ -0,0 +1,1022 @@
+package zio.schema
+
+import zio.random.Random
+import zio.schema.Schema._
+import zio.test._
+
+object DeriveGen {
+
+  def gen[A](implicit schema: Schema[A]): Gen[Sized with Random, A] =
+    schema match {
+      case GenericRecord(structure)             => StructureGen.listMap(structure)(gen(_))
+      case Sequence(schemaA, fromChunk, _)      => Gen.chunkOfBounded(10, 100)(gen(schemaA)).map(fromChunk)
+      case Schema.Enumeration(_ /*structure*/ ) => ??? //TODO
```

Comment:
```suggestion
      case Schema.Enumeration(structure)        => Gen.oneOf(structure.toList.map { case (k, v) => gen(v).map(k -> _) }: _*)
```

Reply:
@igrebenik Isn't it just this?

Reply:
Yes, it is!!! Thank you Adam! Oh my, I spent quite some time to implement this one and couldn't convince the compiler no matter what I did 😅